packages/zoltan2/src/algorithms/Zoltan2_Algorithm.hpp

```diff
@@ -208,6 +294,7 @@ class Algorithm {
 
 
 private:
+  std::vector<partitionTreeNode<part_t> > partitionTree;
```

Comment:
The partitionTree should not be stored in the Algorithm base class.  Users access it from the PartitioningSolution, not from the Algorithm.
 
One option would be to store a partitionTree in the PartitioningSolution, as we store partition information.  Each algorithm would be responsible for setting the partitionTree in the PartitioningSolution, if relevant. 

Another option (and the one used for boxAssign, pointAssign, etc.) is to let specific algorithms store the needed information in their most natural format, and then provide a method to answer getPartitionTree when the PartitionSolution asks for it.  That way, the algorithms wouldn't have to convert their natural info (e.g., rcb_tree treeptr in RCB) to a specific format (partitionTreeNode) and then convert it again to the array format for getPartitionTree.  In this case, only the interface needed by PartitioningSolution would  be in the Algorithm base class;  specific implementation of that interface would be provided in specific algorithms that support partitionTree access.  Then, for example, RCB could fill the arrays directly from its rcb_tree treeptr.